src/SmartComponents/SubscriptionsUtilized/SubscriptionsUtilizedCard.js

```diff
@@ -1,32 +1,138 @@
+import Immutable from 'seamless-immutable';
+import { Tooltip, TooltipPosition } from '@patternfly/react-core';
+import { connect } from 'react-redux';
+import { injectIntl } from 'react-intl';
 import { TemplateCard, TemplateCardBody, TemplateCardHeader } from '../../PresentationalComponents/Template/TemplateCard';
+import Loading from '../../PresentationalComponents/Loading/Loading';
 import { ProgressTemplate } from '../../../../insights-dashboard/src/ChartTemplates/Progress/ProgressTemplate';
-import { injectIntl } from 'react-intl';
 import messages from '../../Messages';
-import PropTypes from 'prop-types';
+import * as AppActions from '../../AppActions';
+import { setRangedDateTime, filterChartData } from './SubscriptionsUtilizedHelpers';
+import { RHSM_API_RESPONSE_DATA, RHSM_API_RESPONSE_DATA_TYPES, RHSM_API_PRODUCT_ID_TYPES, RHSM_API_QUERY_GRANULARITY_TYPES } from './Constants';
 
 /**
  * Subscriptions utilized card for showing the portion of Subscriptions used.
  */
-const SubscriptionsUtilizedCard = ({ intl }) => {
-    return <TemplateCard appName='SubscriptionsUtilized'>
-        <TemplateCardHeader subtitle={ intl.formatMessage(messages.subscriptionsUtilized) }/>
-        <TemplateCardBody>
-            <ProgressTemplate
-                title="Red Hat JBoss"
-                value="0"
-            />
-            <ProgressTemplate
-                title="Red Hat Openshift"
-                value="0"
-                variant="danger"
-            />
-        </TemplateCardBody>
-    </TemplateCard>;
+const SubscriptionsUtilizedCard = ({ intl, subscriptionsUtilizedProductOne, subscriptionsUtilizedProductOneFetch,
+    subscriptionsUtilizedProductOneFetchStatus, subscriptionsUtilizedProductTwo, subscriptionsUtilizedProductTwoFetch,
+    subscriptionsUtilizedProductTwoFetchStatus }) => {
+
+    const [products, setProducts] = useState([]);
+
+    useEffect(() => {
+        const { startDate, endDate } = setRangedDateTime();
+        const options = {
+            granularity: RHSM_API_QUERY_GRANULARITY_TYPES.DAILY,
+            beginning: startDate.toISOString(),
+            ending: endDate.toISOString()
+        };
+
+        subscriptionsUtilizedProductOneFetch(RHSM_API_PRODUCT_ID_TYPES.OPENSHIFT, options);
+        subscriptionsUtilizedProductTwoFetch(RHSM_API_PRODUCT_ID_TYPES.RHEL, options);
+    }, [
+        subscriptionsUtilizedProductOneFetch,
+        subscriptionsUtilizedProductTwoFetch
+    ]);
+
+    useEffect(() => {
+        const chartData = { productOne: {}, productTwo: {} };
+
+        if (subscriptionsUtilizedProductOneFetchStatus === 'fulfilled' || subscriptionsUtilizedProductTwoFetchStatus === 'fulfilled') {
+            const [productOneReport = {}, productOneCapacity = {}] = Immutable.asMutable(subscriptionsUtilizedProductOne, { deep: true }) || [];
+            const [productTwoReport = {}, productTwoCapacity = {}] = Immutable.asMutable(subscriptionsUtilizedProductTwo, { deep: true }) || [];
```

Comment:
If the `asMutable` is an issue, cloning the list/array would be another way to handle it.

Reply:
🆗 as is by me 👍

Reply:
🆗 by me 👍